src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyLineWrappingInBlockCheck.java

```diff
@@ -0,0 +1,323 @@
+            log(ast, MSG_EMPTY_LINE_AFTER);
+        }
+        if (hasEmptyLineBefore(ast)) {
+            if (topSeparator == EmptyLineWrappingInBlockOption.NO_EMPTY_LINE) {
+                log(ast, MSG_NO_EMPTY_LINE_BEFORE);
+            }
+        }
+        else if (topSeparator == EmptyLineWrappingInBlockOption.EMPTY_LINE) {
+            log(ast, MSG_EMPTY_LINE_BEFORE);
+        }
+    }
+
+    /**
+     * Checks whether an empty/blank line is present before the given block.
+     *
+     * @param ast the current node
+     * @return true if an empty/blank line is present before the given block.
+     */
+    private boolean hasEmptyLineBefore(DetailAST ast) {
+        boolean result = false;
+        int index = ast.getLineNo() - 2;
+        if (index > 0) {
+            final FileContents fileContents = getFileContents();
+            String line = fileContents.getLine(index);
+
+            // We need to iterate backwards to find the previous non-commented line.
+            if(line.trim().endsWith(MULTILINE_COMMENT_END)
+                || line.trim().startsWith(INLINE_COMMENT_START)) {
+                // We need to step backwards until we find the start of a multiline comment
+                // or inline comments
+                while((!CommonUtil.isBlank(line)
+                      && (!line.trim().startsWith(MULTILINE_COMMENT_START)
+                      || line.trim().charAt(0) == '*'))
+                      || line.trim().startsWith(INLINE_COMMENT_START)) {
+                    index--;
+                    line = fileContents.getLine(index);
+                }
+                if(line.trim().startsWith(MULTILINE_COMMENT_START)) {
+                    index--;
+                    line = fileContents.getLine(index);
+                }
+            }
+            result = CommonUtil.isBlank(line);
+        }
+        return result;
+    }
+
+    /**
+     * Checks whether an empty/blank line is present after the given block.
+     *
+     * @param ast the current node
+     * @return true if an empty/blank line is present after the given block.
+     */
+    private boolean hasEmptyLineAfter(DetailAST ast) {
+        boolean result = false;
+        final DetailAST rightCurly = getRightCurly(ast);
+        final FileContents fileContents = getFileContents();
+        final int index = rightCurly.getLineNo();
+        if (index < fileContents.getLines().length) {
+            final String text = fileContents.getLine(index);
+            if (text.length() == 0) {
+                result = true;
+            }
+            else {
+                result = fileContents.lineIsBlank(index);
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Finds the respective right curly AST node of the given block.
+     *
+     * @param detailAST the current node
+     * @return the right curly AST node of the given block.
+     */
+    private DetailAST getRightCurly(DetailAST detailAST) {
```

Comment:
It is a bit hader. Here is the code from a similar check:
```
    /**
     * Finds the right curly brace of a code block associated with the specified token.
     *
     * @param ast the token to process
     * @return the token for the left curly brace if found, {@code null} otherwise
     */
    private static DetailAST findRightCurly(DetailAST ast)  {
        DetailAST block = null;
        switch (ast.getType()) {
        case TokenTypes.ANNOTATION_ARRAY_INIT:
        case TokenTypes.ARRAY_INIT:
            block = ast;
            break;

        case TokenTypes.INTERFACE_DEF:
        case TokenTypes.CLASS_DEF:
        case TokenTypes.ANNOTATION_DEF:
        case TokenTypes.ENUM_DEF:
        case TokenTypes.ENUM_CONSTANT_DEF:
        case TokenTypes.LITERAL_NEW:
            block = ast.findFirstToken(TokenTypes.OBJBLOCK);
            break;

        case TokenTypes.LITERAL_SWITCH:
            block = ast;
            break;

        case TokenTypes.LITERAL_ELSE:
            // Special handling for the "else if" case.
            final DetailAST candidate = ast.getFirstChild();
            if (candidate.getType() == TokenTypes.SLIST) {
                block = candidate;
            }
            break;

        case TokenTypes.SLIST:
            // The scope block is a SLIST with another SLIST as the parent.
            if (ast.getParent().getType() == TokenTypes.SLIST) {
                block = ast;
            }
            break;

        case TokenTypes.CASE_GROUP:
            // The CASE_GROUP is the parent of 'case', 'default' and SLIST
            // parent.
            block = ast.getLastChild().findFirstToken(TokenTypes.SLIST);
            break;

        case TokenTypes.LITERAL_CASE:
        case TokenTypes.LITERAL_DEFAULT:
            // The 'case' and 'default' are siblings for SLIST parent.
            block = ast.getParent().getLastChild().findFirstToken(TokenTypes.SLIST);
            break;

        default:
            // All the rest: for/while/if/STATIC_INIT,lambda,etc.
            block = ast.findFirstToken(TokenTypes.SLIST);
            break;
        }

        // The right curly brace is always the last child.
        DetailAST rightCurly = null;
        if (block != null) {
            rightCurly = block.getLastChild();
        }

        return rightCurly;
    }
```

Reply:
Done